sched/scheduler/stateful_scheduler_test.go

```diff
@@ -311,11 +321,14 @@ func Test_StatefulScheduler_JobRunsToCompletion(t *testing.T) {
 	jobId, _ := s.ScheduleJob(jobDef)
 
 	// add additional saga data
-	sagaLogMock.EXPECT().LogMessage(saga.MakeStartTaskMessage(jobId, taskId, nil))
-	sagaLogMock.EXPECT().LogMessage(TaskMessageMatcher{Type: &sagaStartTask, JobId: "job1", TaskId: "task1", Data: gomock.Any()}).MaxTimes(1)
-	endMessageMatcher := TaskMessageMatcher{JobId: jobId, TaskId: taskId, Data: gomock.Any()}
-	sagaLogMock.EXPECT().LogMessage(endMessageMatcher)
-	sagaLogMock.EXPECT().LogMessage(saga.MakeEndSagaMessage(jobId))
+	gomock.InOrder(
```

Comment:
The gomock.Any() was at times catching the nil case preceding it. This was a bear to figure out because gomock doesn't appear in the stack traces by default so I thought it was a sched locking issue...